torchvision/models/detection/anchor_utils.py

```diff
@@ -132,27 +130,17 @@ def grid_anchors(self, grid_sizes: List[List[int]], strides: List[List[Tensor]])
 
         return anchors
 
-    def cached_grid_anchors(self, grid_sizes: List[List[int]], strides: List[List[Tensor]]) -> List[Tensor]:
```

Comment:
I hate to say this but this is technically public 😢 so I'm not sure we can remove it.

Maybe we should keep it and just return `self.grid_anchors(grid_sizes, strides)`. The now-confusing naming could be explained with a comment indicating we removed caching but we're keeping the method for backward compat.

Reply:
Should I put a deprecation warning in this function for now, and remove it in the next release?

Reply:
We can add a warning, but from past discussions with @fmassa I don't think torchvision ever removes stuff unless it's a bug

Reply:
The overall encapsulation is very problematic in many places of the code as Nicolas pointed out. Personally I would be OK with a BC-break here. I can't imagine many valid usages of someone calling this obviously internal method directly. Can't find any such call on Github or on FBcode.

Reply:
I see some deprecation warnings present in torchvision([example](https://github.com/pytorch/vision/blob/54d88f79b44747e462dee7523fa17d24116bd783/torchvision/transforms/functional_tensor.py#L236)). I know in torchaudio, we deprecated a number of functions/methods and there is a [decorator](https://github.com/pytorch/audio/blob/b5d802795ffb381947e9fbbc1b1da40aa6c64b5f/torchaudio/_internal/module_utils.py#L42) to elegantly do this, may be something we could do as well.

Reply:
I wouldn't bother with keeping BC here -- this is not something that I would expect anyone to be using really.

But if you want to raise a warning with a deprecation cycle I'm ok with this as well